crates/sui-adapter/src/programmable_transactions/execution.rs

```diff
@@ -1084,6 +1083,39 @@ pub enum SpecialArgumentLayout {
     Ascii,
     /// A UTF8 encoded string
     UTF8,
+    // needed for Option validation
```

Comment:
@bmwill just the top commit here, but I missed checking `Option` for non-string types, whoops

Let me know if you have thoughts to make this simpler

Reply:
do we not need to worry about checking for arbitrary struct layouts?

Reply:
No, because we only allow "primitives", which then is extended to include `object::ID`, `String` and `Option`

Reply:
except for my genesis stuff ;)